Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealityInspectorUtility.cs

```diff
@@ -453,5 +454,130 @@ public static Quaternion RotationHandle(Object target, Vector3 position, Quatern
         }
 
         #endregion Handles
+
+        #region Profiles
+
+        private static readonly GUIContent NewProfileContent = new GUIContent("+", "Create New Profile");
+        private static Dictionary<Object, UnityEditor.Editor> profileEditorCache = new Dictionary<Object, UnityEditor.Editor>();
+
+        /// <summary>
+        /// Draws an editor for a profile object.
+        /// </summary>
+        /// <param name="objectReferenceValue"></param>
+        public static void DrawSubProfileEditor(Object profileObject, bool renderProfileInBox)
+        {
+            if (profileObject == null)
+            {
+                return;
+            }
+
+            UnityEditor.Editor subProfileEditor = null;
+            if (!profileEditorCache.TryGetValue(profileObject, out subProfileEditor))
+            {
+                subProfileEditor = UnityEditor.Editor.CreateEditor(profileObject);
+                profileEditorCache.Add(profileObject, subProfileEditor);
+            }
+
+            // If this is a default MRTK configuration profile, ask it to render as a sub-profile
+            if (typeof(BaseMixedRealityToolkitConfigurationProfileInspector).IsAssignableFrom(subProfileEditor.GetType()))
+            {
+                BaseMixedRealityToolkitConfigurationProfileInspector configProfile = (BaseMixedRealityToolkitConfigurationProfileInspector)subProfileEditor;
+                configProfile.RenderAsSubProfile = true;
+            }
+
+            var subProfile = profileObject as BaseMixedRealityProfile;
+            if (subProfile != null && !subProfile.IsCustomProfile)
+            {
+                EditorGUILayout.HelpBox("Clone this default profile to edit properties below", MessageType.Warning);
+            }
+
+            if (renderProfileInBox)
+            {
+                EditorGUILayout.BeginVertical(EditorStyles.helpBox);
```

Comment:
switch maybe to switch to new EditorGUILayout.VerticalScope(EditorStyles.HelpBox)
and use ? : operator to pass correct style in?